### ISSUE TITLE
Stuck when adding MetSys.set_player_position(global_position)

### ISSUE THREAD

**Issue by H0W2D017** [KoBeWi/Metroidvania-System#4]
For some reason i get stuck at line 92 when i add the MetSys.set_player_position(global_position)

`func visit_cell(coords: Vector3i):  save_data.explore_cell(coords)`

with 

> Invalid call. Nonexistent function 'explore_cell' in base Nil

so i tought that i would check the sample project but i cant get that one to run ither.

I am using Godot 4.2 beta 4

**Comment by KoBeWi** [comment]
You need to initialize the save data, by using `MetSys.set_save_data()`.

> i tought that i would check the sample project but i cant get that one to run ither.

That's weird. Do you get the same error?

**Comment by H0W2D017** [comment]
Awsome, that did the trick.
No when i fire up the test project i am only getting a blank screen with the UI on it.
I am loading the scene named "Game"

**Comment by KoBeWi** [comment]
That's weird, is the project umodified? Do you get *any* errors? Is the plugin enabled?

**Comment by H0W2D017** [comment]
**This only have to do with running the sample project as the issue i had was resolved by using MetSys.set_save_data().**

So if i make a new project and download metsys from the AssetLib i get stuck at line 64:
`map = load(map_path).instantiate()`

but if i download the repo from git and import that as a project it works and loads the test level so it seems to have something to do with downloading it from the AssetLib.

Two things i notice that differ from downloading the asset from Lib and Git is that when i download it from Lib the "Map Editor" window is completely blank and the "Map Root Folder" is res:// while when downloading the git project the folder is res://SampleProject/Maps/

**Comment by KoBeWi** [comment]
When you download to existing project, the `project.godot` is ignored and it contains some settings related to the addon. So that would explain why it breaks.

**Comment by ngschroeder** [comment]
I also get the `map = load(map_path).instantiate()` error on a new project. Changing the path in the project settings to res://scenes (where my level .tscn file is) doesn't seem to help.

This is an awesome addon, I think a bare-minimum guide on how to start a fresh project with only 1 room and a playable character would be a good addition. The sample project is a nice showcase of features, but no matter how much code I think I'm stripping out correctly, I can't seem to get a simple room with my player in it to run.

**Comment by KoBeWi** [comment]
`load()` is a Godot's function. If your `map_path` comes from MetSys, it's just a file name and it doesn't point to any resource. You should use `get_full_room_path()` to get full loadable scene path. This is explained in the example project.

> I think a bare-minimum guide on how to start a fresh project with only 1 room and a playable character would be a good addition.

Well, the sample project is like that and its README section + code comments explain how to integrate the system. Though maybe I could include more pointers for where to start.

**Comment by nyrrrr** [comment]
> Awsome, that did the trick. No when i fire up the test project i am only getting a blank screen with the UI on it. I am loading the scene named "Game"

I think this is because the function `get_size` of `RoomInstance` is incorrect.
it should be:
```
func get_size() -> Vector2:
	return Vector2(min_cell - max_cell + Vector2i.ONE) * MetSys.settings.in_game_cell_size
```
At least in the version I downloaded from the AssetLib, min_cell and max_cell were swapped.

**Comment by ngschroeder** [comment]
I figured out the stuff I was struggling with, it mostly involved looking for the info needed for the start scene and start position. This is a really great add-on, and all the features are working quite nicely on a fresh project now that I'm past the initial setup. Thanks for sharing this!

**Comment by KoBeWi** [comment]
I added a note in the readme about initializing save data, and also a quick start guide.
Since the original issue here is resolved, I'll close this in favor of #13 If you have any feedback about initial setup please comment in there.
